### PR TITLE
[server] Ensure incremental prebuilds always use the latest config

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

Ensure Incremental Prebuilds always use the latest config (by always overwriting the parent prebuild's config with the most-recently fetched config).

From the in-line comment:

> `createForPrebuiltWorkspace` 1:1 copies the config from the parent prebuild.
> Above, we've made sure that the parent's prebuild tasks (before/init/prebuild) are still the same as now.
> However, other non-prebuild config items might be outdated (e.g. any command task, VS Code extension, ...)
> To fix this, we overwrite the new prebuild's config with the most-recently fetched config.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7475

## How to test

See "Steps to reproduce" in https://github.com/gitpod-io/gitpod/issues/7475

## Release Notes

```release-note
[server] Ensure incremental prebuilds always use the latest config
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankeromnes** after the PR has been reviewed.

Associated issue: *#7475*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by JanKoehnlein — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 8e752d9f45bfd1bbedbd428e3ba47ee9d7403561</details>

**Pull request merged**
